BatchAuctionOptimization/batchauctions.tex

```diff
@@ -854,80 +819,94 @@ \subsection{Mixed-integer linear programming model I}
+
+These two types of inequalities can also be generalized to chains of orders along cycles of tokens.
+Let $ (\tau_{j_1}, \tau_{j_2}, \ldots, \tau_{j_m}, \tau_{j_1}) $ be such a token cycle of
+length $ m $, and let there be a set of orders
+$ \left(
+\omega_{i_1} = (j_1,j_2,\cdot,\cdot,\pi_{i_1}), \>
+\omega_{i_2} = (j_2,j_3,\cdot,\cdot,\pi_{i_2}), \>
+\ldots, \>
+\omega_{i_m} = (j_m,j_1,\cdot,\cdot,\pi_{i_m})
+\right) $.
+Now, if $ \prod_{l=1}^m \pi_{i_l} > 1 $, our arbitrage-freeness requirement again forbids that all
+orders can be executed at the same time. Hence, we can write
 \begin{align}
-  z^{(1)} + z^{(2)} \le 1.
-  \label{eq:mip1_order_exclusion}
+  \sum\limits_{l=1}^m z_{i_l} \le m-1.
+  \label{eq:valid_inequality_B1}
 \end{align}
-This type of inequality can of course also be generalized to chains of orders along cycles of
-tokens.
-Let $ (\tau_{j_1}, \tau_{j_2}, \ldots, \tau_{j_m}, \tau_{j_1}) $ be such a token cycle of length
-$ m $.
-Then, for a set of orders
-$ \{
-\omega^{(1)} = (j_1,j_2,\cdot,\cdot,\pi^{(1)}), \>
-\omega^{(2)} = (j_2,j_3,\cdot,\cdot,\pi^{(2)}), \>
-\ldots, \>
-\omega^{(m)} = (j_m,j_1,\cdot,\cdot,\pi^{(m)})
-\} $
-with $ \prod_{i=1}^m \pi^{(i)} < 1 $, the following needs to hold:
+
+In the other case where $ \prod_{l=1}^m \pi_{i_l} \le 1 $, we know that at least one of the orders
+needs to be allowed being executed, which yields
 \begin{align}
-  \sum\limits_{i=1}^m z^{(i)} \le m-1.
-  \label{eq:mip1_order_exclusion_general}
+  \sum\limits_{l=1}^m z_{i_l} \ge 1.
+  \label{eq:valid_inequality_B2}
 \end{align}
-\end{subequations}
 
-In both cases, there are many possible choices of subsets of orders that are mutually exclusive.
-This holds true in particular for the generalized constraint variant
-\eqref{eq:mip1_order_exclusion_general} the longer the token cycles become.
-Hence, for instances with a large number of orders, it will be impractical to include all possible
-inequalities, so some care should be taken on selecting the most impactful ones.
-So far, we have done experiments with cycle lengths $ m \in \{2,3\} $, in which the mutual
-exclusion constraints have shown to have a significant positive impact on the solving process.
+In both cases \eqref{eq:valid_inequality_B1} and \eqref{eq:valid_inequality_B2}, we can use the
+inequalities \eqref{eq:valid_inequality_A} in order to detect redundancies.
+However, even the numbers of non-redundant inequalities of these types grow significantly with the
+length of the token cycles that are being considered.
+For our practical purposes, we have incorporated inequalities for cycles of length $ m=\{2,3\} $
+and observed remarkable performance gains.
+
+\todo[inline]{Describe how exactly non-redundant inequalities are determined!?}
 
-\todo[inline]{Which inequalities to select!?}
+\todo[inline]{Computational results!}
 
 
 \newpage
 \subsection{Mixed-integer linear programming model II}
 \label{subsec:MIP2}
 
+\todo[inline]{Adjust to only use sell orders!}
```

Comment:
Is this not what has already been done here?

Reply:
I have not worked through model II, yet (since it's not the one I'm using most of the time).

Reply:
Will do some other day...